webAO/sw.js

```diff
@@ -0,0 +1,26 @@
+const cacheName = 'webAO';
+
+// Cache all the files to make a PWA
+self.addEventListener('install', (e) => {
+  e.waitUntil(
+    caches.open(cacheName).then((cache) =>
+      // Our application only has two files here index.html and manifest.json
+      // but you can add more such as style.css as your app grows
+      cache.addAll([
+        './',
+        './index.html',
+        '../manifest.json',
+      ])),
+  );
+});
+
+// Our service worker will intercept all fetch requests
+// and check if we have cached the file
+// if so it will serve the cached file
+self.addEventListener('fetch', (event) => {
+  event.respondWith(
+    caches.open(cacheName)
+      .then((cache) => cache.match(event.request, { ignoreSearch: true }))
+      .then((response) => response || fetch(event.request)),
+  );
+});
```

Comment:
do PWAs really require service workers? i just think service workers are botnets

Reply:
It's a requirement to be considered a PWA. Not sure why. This is pretty boilerplate code for getting it to work.